src/java/voldemort/utils/ExceptionUtils.java

```diff
@@ -0,0 +1,15 @@
+package voldemort.utils;
+
+/**
+ * Static utility functions to deal with exceptions.
+ */
+public class ExceptionUtils {
+    public static boolean recursiveClassEquals(Throwable throwableToInspect, Class throwableToLookFor) {
```

Comment:
I agree that this is better way to do it, but I am little scary of touching the old code. My mindset for making change is , make it, test it in EI, simulate in EI . At this point, I will not touch that code for now , but if some of us touch this in the future, we should fix it.